```
catboost> FAILED: catboost/private/libs/functools/CMakeFiles/private-libs-functools.dir/forward_as_const.cpp.o                                         
catboost> /nix/store/xl0vlc2wdchfbq8536zs19pj2r3xdmma-clang-wrapper-19.1.5/bin/clang++ -DARCADIA_BUILD_ROOT=/build/source/build -DARCADIA_ROOT=/build/s
ource -DCATBOOST_OPENSOURCE=yes -I/build/source -I/build/source/build -I/build/source/contrib/libs/linux-headers -I/build/source/contrib/libs/linux-hea
ders/_nf -I/build/source/contrib/libs/cxxsupp/libcxx/include -I/build/source/contrib/libs/cxxsupp/libcxxrt/include -I/build/source/contrib/libs/clang16
-rt/include -I/build/source/contrib/libs/zlib/include -I/build/source/contrib/libs/double-conversion -I/build/source/contrib/libs/libc_compat/include/r
eadpassphrase -I/build/source/contrib/libs/libc_compat/reallocarray -I/build/source/contrib/libs/libc_compat/random -fexceptions   -fno-common   -fcolo
r-diagnostics   -faligned-allocation   -fdebug-default-version=4   -ffunction-sections   -fdata-sections   -Wall   -Wextra   -Wno-parentheses   -Wno-im
plicit-const-int-float-conversion   -Wno-unknown-warning-option   -pipe   -D_THREAD_SAFE   -D_PTHREADS   -D_REENTRANT   -D_LARGEFILE_SOURCE   -D__STDC_
CONSTANT_MACROS   -D__STDC_FORMAT_MACROS   -D__LONG_LONG_SUPPORTED  -D_GNU_SOURCE -DLIBCXX_BUILDING_LIBCXXRT -fuse-init-array -D_FILE_OFFSET_BITS=64  -
m64    -msse2     -msse3     -mssse3         -msse4.1       -msse4.2       -mpopcnt      -mcx16 -D_YNDX_LIBUNWIND_ENABLE_EXCEPTION_BACKTRACE   -Woverlo
aded-virtual   -Wimport-preprocessor-directive-pedantic   -Wno-undefined-var-template   -Wno-return-std-move   -Wno-defaulted-function-deleted   -Wno-p
essimizing-move   -Wno-deprecated-anon-enum-enum-conversion   -Wno-deprecated-enum-enum-conversion   -Wno-deprecated-enum-float-conversion   -Wno-ambig
uous-reversed-operator   -Wno-deprecated-volatile      -DSSE_ENABLED=1     -DSSE3_ENABLED=1     -DSSSE3_ENABLED=1         -DSSE41_ENABLED=1       -DSSE
42_ENABLED=1       -DPOPCNT_ENABLED=1      -DCX16_ENABLED=1 -O3 -DNDEBUG -std=c++20 -fPIC -DLIBCXX_BUILDING_LIBCXXRT -nostdinc++ -D_libunwind_ -MD -MT 
catboost/private/libs/functools/CMakeFiles/private-libs-functools.dir/forward_as_const.cpp.o -MF catboost/private/libs/functools/CMakeFiles/private-lib
s-functools.dir/forward_as_const.cpp.o.d -o catboost/private/libs/functools/CMakeFiles/private-libs-functools.dir/forward_as_const.cpp.o -c /build/sour
ce/catboost/private/libs/functools/forward_as_const.cpp                                                                                                
catboost> In file included from /build/source/catboost/private/libs/functools/forward_as_const.cpp:1:                                                  
catboost> /build/source/catboost/private/libs/functools/forward_as_const.h:107:42: error: a template argument list is expected after a name prefixed by
 the template keyword [-Wmissing-template-arg-list-after-template-kw]                                                                                  
catboost>   107 |                 return TParent::template ForwardSelfAsConst(                                                                         
catboost>       |                                          ^                                                                                           
catboost> 1 error generated. 
```

--- a/pkgs/development/python-modules/orange3/default.nix
+++ b/pkgs/development/python-modules/orange3/default.nix
@@ -1,52 +1,58 @@
 {
   lib,
-  gitUpdater,
   stdenv,
-  baycomp,
-  bottleneck,
   buildPythonPackage,
-  chardet,
+  fetchFromGitHub,
+
+  # build-system
+  oldest-supported-numpy,
+  setuptools,
+
+  # nativeBuildInputs
   copyDesktopItems,
   cython,
+  qt5,
+  recommonmark,
+  sphinx,
+
+  # dependencies
+  baycomp,
+  bottleneck,
   catboost,
-  xgboost,
-  fetchFromGitHub,
-  fetchurl,
+  chardet,
   httpx,
   joblib,
   keyring,
   keyrings-alt,
-  makeDesktopItem,
   matplotlib,
   numpy,
-  oldest-supported-numpy,
   openpyxl,
   opentsne,
   orange-canvas-core,
   orange-widget-base,
   pandas,
-  pytestCheckHook,
-  pytest-qt,
-  pyqtgraph,
-  pyqt5,
-  pyqtwebengine,
-  python,
-  python-louvain,
-  pythonOlder,
-  pyyaml,
   pip,
-  qt5,
+  pyqt5,
+  pyqtgraph,
+  pyqtwebengine,
+  python-louvain,
+  pyyaml,
   qtconsole,
-  recommonmark,
   requests,
   scikit-learn,
   scipy,
   serverfiles,
-  setuptools,
-  sphinx,
-  wheel,
+  xgboost,
   xlrd,
   xlsxwriter,
+
+  makeDesktopItem,
+
+  # passthru
+  gitUpdater,
+  python,
+  pytest-qt,
+  pytestCheckHook,
 }:
 
 let
@@ -55,14 +61,17 @@ let
     version = "3.37.0";
     pyproject = true;
 
-    disabled = pythonOlder "3.7";
-
     src = fetchFromGitHub {
       owner = "biolab";
       repo = "orange3";
       tag = version;
       hash = "sha256-3PybiHXH6oIYJb78/a7LnQA6cYUicDx4Mf65QhIhT4w=";
     };
+
+    build-system = [
+      oldest-supported-numpy
+      setuptools
+    ];
 
     postPatch = ''
       substituteInPlace pyproject.toml \
@@ -74,50 +83,47 @@ let
 
     nativeBuildInputs = [
       copyDesktopItems
-      oldest-supported-numpy
       cython
       qt5.wrapQtAppsHook
       recommonmark
-      setuptools
       sphinx
-      wheel
     ];
 
     enableParallelBuilding = true;
 
     pythonRelaxDeps = [ "scikit-learn" ];
 
-    propagatedBuildInputs = [
-      numpy
-      scipy
-      chardet
+    dependencies = [
+      baycomp
+      bottleneck
       catboost
-      xgboost
+      chardet
+      httpx
+      joblib
+      keyring
+      keyrings-alt
+      matplotlib
+      numpy
       openpyxl
       opentsne
-      qtconsole
-      setuptools
-      bottleneck
-      matplotlib
-      joblib
-      requests
-      keyring
-      scikit-learn
-      pandas
-      pyqtwebengine
-      serverfiles
       orange-canvas-core
+      orange-widget-base
+      pandas
+      pip
+      pyqt5
+      pyqtgraph
+      pyqtwebengine
       python-louvain
+      pyyaml
+      qtconsole
+      requests
+      scikit-learn
+      scipy
+      serverfiles
+      setuptools
+      xgboost
       xlrd
       xlsxwriter
-      httpx
-      pyqtgraph
-      pyqt5
-      orange-widget-base
-      keyrings-alt
-      pyyaml
-      baycomp
-      pip
     ];
 
     # FIXME: ImportError: cannot import name '_variable' from partially initialized module 'Orange.data' (most likely due to a circular import) (/build/source/Orange/data/__init__.py)
@@ -181,13 +187,13 @@ let
           chmod +w -R .
 
           substituteInPlace Orange/classification/tests/test_xgb_cls.py \
-            --replace test_learners mk_test_learners
+            --replace-fail test_learners mk_test_learners
 
           substituteInPlace Orange/modelling/tests/test_xgb.py \
-            --replace test_learners mk_test_learners
+            --replace-fail test_learners mk_test_learners
 
           substituteInPlace Orange/**/tests/*.py \
-            --replace test_filename filename_test
+            --replace-fail test_filename filename_test
 
           # TODO: debug why orange is crashing on GC, may be a upstream issue
           chmod +x Orange/__init__.py
@@ -196,8 +202,8 @@ let
         '';
 
         nativeBuildInputs = [
-          pytestCheckHook
           pytest-qt
+          pytestCheckHook
         ];
 
         postCheck = ''

